### ISSUE TITLE
protobuf 4 is not supported

### ISSUE THREAD

**Issue by aucampia** [tensorflow/tensorflow#59221]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Build/Install

### Have you reproduced the bug with TF nightly?

Yes

### Source

binary

### Tensorflow Version

2.11.0

### Custom Code

No

### OS Platform and Distribution

Fedora Linux 37 (Xfce)

### Python version

3.11

### Current Behaviour?

```shell
When I try to use tensorflow with protobuf newer than version 3 I get this:


$ poetry add 'tensorflow==2.11.0' 'protobuf>=4'

Updating dependencies
Resolving dependencies... (0.2s)

Because tensorflow-issue depends on tensorflow (2.11.0) which depends on protobuf (>=3.9.2,<3.20), protobuf is required.
So, because tensorflow-issue depends on protobuf (>=4), version solving failed.


The reason for this is:

https://github.com/tensorflow/tensorflow/blame/d0dc98c5c8f5deea2447a3405af20f5fb245a561/tensorflow/tools/pip_package/setup.py#L100-L107

My expectation was that tensorflow supports more recent versions of protobuf.
```

### Standalone code to reproduce the issue

```shell
mkdir -vp /var/tmp/tensorflow-issue
poetry -C /var/tmp/tensorflow-issue init --no-interaction
poetry -C /var/tmp/tensorflow-issue add 'tensorflow==2.11.0' 'protobuf>=4'
```

### Relevant log output

_No response_</details>

**Comment by aucampia** [comment]
This is affecting me because I use apache beam:

- https://github.com/apache/beam/issues/24569

**Comment by dwyatte** [comment]
Any update here?

This change appears to have been introduced in TensorFlow 2.10.x. It was previously just `protobuf (>=3.9.2)` in 2.9.x without the upper bound of <3.20.

I'm most interested in getting 3.20 support since there was a [breaking change](https://github.com/protocolbuffers/protobuf/releases/tag/v3.20.0) around how code generation happens, although there are only three releases after that in 3.x so 4.x support would be welcome too.

**Comment by AnandInguva** [comment]
Is Tensorflow 2.12 gonna support Protobuf>=4.21.0?

**Comment by dwyatte** [comment]
> Is Tensorflow 2.12 gonna support Protobuf>=4.21.0?

> I am working on updating protobuf dependency in Apache Beam but Apache Beam has a dependency on Tensorflow. So we are not able to make the update of Protobuf on tensorflow==2.11.0

I don't work on TensorFlow, so can't say with any authority, but I downloaded the [release candidate](https://pypi.org/project/tensorflow/2.12.0rc0/) and it appears to have changed the protobuf dependency to at least go beyond 3.20

2.11: Requires-Dist: protobuf (<3.20,>=3.9.2)
2.12rc0: Requires-Dist: protobuf (!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0dev,>=3.20.3)

**Comment by mihaimaruseac** [comment]
That is exactly the plan, after 84f40925e929d05e72ab9234e53c729224e3af38

**Comment by michalmodras** [comment]
@mihaimaruseac Do you know when Protobuf 4.x support will land in Tensorflow? As relevant to apache-beam python library, it is also blocking upgrade in Apache Airflow.

**Comment by mihaimaruseac** [comment]
It should be available in TF 2.12.0-RC1 (TF 2.12.0 when the release finishes)

**Comment by michalmodras** [comment]
And when do we expect the 2.12.0 release to finish?

**Comment by mihaimaruseac** [comment]
The release process for TF usually has 2 RCs (0 and 1) and then a final release. The process for the next release starts a week after the previous RC finished.

So, likely next week. But delays could always occur.

**Comment by michalmodras** [comment]
Great, thank you for the update!

**Comment by eapolinario** [comment]
https://github.com/tensorflow/tensorflow/commit/84f40925e929d05e72ab9234e53c729224e3af38 landed in 2.12.0. This ticket can now be closed, right?
